aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+    }
+
+    /// Creates a new multisig account with the specified additional owner list and signatures required.
+    ///
+    /// @param additional_owners The owner account who calls this function cannot be in the additional_owners and there
+    /// cannot be any duplicate owners in the list.
+    /// @param signatures_require The number of signatures required to execute a transaction. Must be at least 1 and
+    /// at most the total number of owners.
+    public entry fun create_with_owners(owner: &signer, additional_owners: vector<address>, signatures_required: u64) {
+        let (multisig_account, multisig_signer_cap) = create_multisig_account(owner);
+        vector::push_back(&mut additional_owners, address_of(owner));
+        create_with_owners_internal(
+            &multisig_account,
+            additional_owners,
+            signatures_required,
+            option::some(multisig_signer_cap),
+        );
+    }
+
+    fun create_with_owners_internal(
+        multisig_account: &signer,
+        owners: vector<address>,
+        signatures_required: u64,
+        multisig_account_signer_cap: Option<SignerCapability>,
+    ) {
+        assert!(
+            signatures_required > 0 && signatures_required <= vector::length(&owners),
+            error::invalid_argument(EINVALID_SIGNATURES_REQUIRED),
+        );
+
+        validate_owners(&owners, address_of(multisig_account));
+        move_to(multisig_account, MultisigAccount {
+            owners,
+            signatures_required,
+            transactions: table::new<u64, MultisigTransaction>(),
+            // First transaction will start at id 1 instead of 0.
+            last_transaction_id: 0,
+            next_transaction_id: 1,
+            signer_cap: multisig_account_signer_cap,
+            add_owners_events: new_event_handle<AddOwnersEvent>(multisig_account),
+            remove_owners_events: new_event_handle<RemoveOwnersEvent>(multisig_account),
+            update_signature_required_events: new_event_handle<UpdateSignaturesRequiredEvent>(multisig_account),
+            create_transaction_events: new_event_handle<CreateTransactionEvent>(multisig_account),
+            approve_transaction_events: new_event_handle<ApproveTransactionEvent>(multisig_account),
+            reject_transaction_events: new_event_handle<RejectTransactionEvent>(multisig_account),
+            remove_transaction_events: new_event_handle<RemoveTransactionEvent>(multisig_account),
+            execute_transaction_events: new_event_handle<TransactionExecutionSucceededEvent>(multisig_account),
+            transaction_execution_failed_events: new_event_handle<TransactionExecutionFailedEvent>(multisig_account),
+        });
+    }
+
+    /// Add new owners to the multisig account. This can only be invoked by the multisig account itself, through the
+    /// proposal flow.
+    ///
+    /// Note that this function is not public so it can only be invoked directly instead of via a module or script. This
```

Comment:
Could you please give a reference where how the transaction can be executed bypassing the public entry function check?

Reply:
No specific reference. If this is public entry, another module being called as part of a multisig tx can call multisig_account::add_owners with the signer of the multisig and maliciously alters the owner list.

Reply:
Yep! Loving the protection you included here. I am wondering how rust is bypassing the public visibility check? Was not able to find the code 🤣

Reply:
The VM can invoke a function regardless of its visibility. Example in existing code: https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/aptos-vm/src/aptos_vm.rs#L467

Reply:
Is it possible that it may invoke some unexpected protected function?